### PR TITLE
fix: Attach `credentialId` to location option

### PR DESCRIPTION
**Pull request by joeauyeung**
## What does this PR do?

This PR adds a missing field `credentialId` to event type location options. This credentialId is expected in `handleNewBooking` under `conferenceCredentialId`. For team events, the default host is not always present but it was still falling back to their location option.

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- On the event type setup tab, the location options should contain the `credentialId`
- Install a video app that is scoped to a team
- Set that team installed location to a team event type
- When creating a new booking the `conferenceCredentialId` in `handleNewBooking` should be defined

## Mandatory Tasks

- [ ] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes

### CODE REVIEW

File: apps/web/components/eventtype/EventSetupTab.tsx [hunk 1]
```diff
@@ -382,7 +382,7 @@ export const EventSetupTab = (
                       !validLocations.find((location) => location.type === newLocationType);
 
                     if (canAppendLocation) {
-                      append({ type: newLocationType });
+                      append({ type: newLocationType, credentialId: e.credentialId });
```
Comment (Udit-takkar) [inline apps/web/components/eventtype/EventSetupTab.tsx:385]: I think we should use something like

  ```
 append({
                        type: newLocationType,
                        ...(e.credential && {
                          credentialId: e.credential.id,
                          teamName: e.credential.team?.name,
                        }),
                      });
```

**Review by Udit-takkar — CHANGES_REQUESTED** [review]
@joeauyeung we should also pass teamName like in https://github.com/calcom/cal.com/blob/51fd4102ae15af67e973257e996017d20fc309bb/apps/web/components/dialog/EditLocationDialog.tsx#L360


and we should probably use type credentialId and teamName instead of only credentialId as both credential.id and credential.team.name would be required in EditLocationDialog.tsx and EventSetup.

**Review by Udit-takkar — APPROVED** [review]
LGTM. I Fixed everything now

**Comment by joeauyeung** [comment]
> @joeauyeung we should also pass teamName like in
> 
> https://github.com/calcom/cal.com/blob/51fd4102ae15af67e973257e996017d20fc309bb/apps/web/components/dialog/EditLocationDialog.tsx#L360
> 
> and we should probably use type credentialId and teamName instead of only credentialId as both credential.id and credential.team.name would be required in EditLocationDialog.tsx and EventSetup.

Good call thank you!

**Review by CarinaWolli — APPROVED** [review]
LGTM!

**Pull request merged**